ci/jenkins/jobs/macros.yaml

```diff
@@ -231,6 +231,7 @@
           #!/bin/bash
           set -ex
           DOCKER_REGISTRY="$(head -n1 ci/docker-registry)"
+          kind delete clusters --all
```

Comment:
By default kind delete clusters --all, By default, it uses the ~/.kube/config file to delete all the clusters, which may not be the right file if you have multiple kubeconfig files for different clusters.

Reply:
yes, as far as i know, all the configs are presents in the general kubeconfig file, but i think there may be an issue if while creating the cluster we specify the kubeconfig parameter and export it to some other location, then in that case the config may not be saved to the .kube/config file. @KMAnju-2021 can you test this once. @XinShuYang what's your opinion on this.

Reply:
@rajnkamr Current CI scipt export kubeconfig after cluster creation. In this case the cluster info is still saved in default kubeconfig.

Reply:
@XinShuYang   , if all clusters entries are added to same kubeconfig file during creation then it will remain there and "kind delete clusters --all" deletes all cluster using default kubeconfig file. @jainpulkit22 Exporting from default kubeconfig to other location does not wipe out the default kubeconfig. "the case which is problematic is when different kubconfig files are provided during each cluster creation" , in this scenario all created cluster configs for cluster is going to same kubeconfig file.

Reply:
Hi All, yesterday I got some suggestion from zhengsheng, We need to modify the current logic to delete the kind cluster because current cmd "kind delete clusters --all" will break the parallel runs in single VM. working on logic, thanks!

Reply:
@KMAnju-2021  Need to address delete clusters case in parallel job run scenario

Reply:
Hi @KMAnju-2021, it's better to consider supporting parallel job in cleanup function. After discussing with Zhengsheng, we found a simpler solution that identifies kind cluster running for too long at the beginning of kind jobs. By setting a timeout value, such as 4 hours, we can conditionally delete clusters. 
After completing each job, it's only necessary to delete the cluster used for that specific job, which is consistent with the current implementation. 
This approach eliminates the need for creating a new gc job, making it more convenient to manage and deploy..

Reply:
@XinShuYang 
Rather you could run a cron job , So have a Jenkins job that runs on a regular schedule, 
[may be daily(not sure if 4 hour is enough to assume that the related cluster needs to be deleted or has been aborted) ] 
This CronJob will delete the Kind clusters based on their age. 
so create a CronJob that runs a script to delete Kind clusters older than a specified age. The script can use the kind command-line tool to list and delete Kind clusters, and the date command to compare the age of the clusters. 

{Additionally You could try to use the Jenkins Kubernetes plugin to create and manage Kubernetes resources from Jenkins.}

Reply:
@XinShuYang ,  
 if we can set "abort job value" to 2 hour, then it should be fine to choose a higher value than 2 hours 

@KMAnju-2021 Implementation wise, we might want to mention precisely like say we can add to the cleanup function that will delete all hanging kind clusters if the jobs/clusters are timeout. Ideally to check this you can loop through each kind cluster and check the timeout value wrt cluster creation timestamp and then delete each one

Reply:
@XinShuYang @rajnkamr Thanks for the suggestions, will modify the code

Reply:
@KMAnju-2021 
As discussed, we need to take care of following
1.Currently only conformance and network policy tests are running in kind, we might be adding e2e test or any other tests as well in future, so we will have to use a common cleanup function which can be used and called for cleanup

2. To handle parallel run, you might want to have queue which might contain all running jobs and implement a logic around queue which deletes clusters when timer expires for each job while starting a new job this function will get invoked and try to clean up the expired job clusters(if any)